akka-stream/src/main/scala/akka/stream/impl/Buffers.scala

```diff
@@ -332,6 +332,11 @@ private[impl] final class PartitionedBuffer[K, V](size: Int) {
     }
   }
 
+  def usedInPartition(key: K): Int =
+    if (partitionBuffers.contains(key)) {
+      partitionBuffers(key).used
+    } else 0
```

Comment:
might be slightly more efficient to `match`
```
partitionBuffers.get(key) match {
  case Some(v) => v.used
  case None => 0
}

Reply:
Torn between hash lookup vs. extra allocation, since this gets called in a tight loop.  Is the general experience that the allocation and match is faster than double lookup?

Reply:
`contains` and `partitionBuffers(key)` are also allocating `Some`, i.e. two allocations and two hash lookup